src/components/SearchModal/PairSearch.tsx

```diff
@@ -0,0 +1,143 @@
+import { Pair } from 'dxswap-sdk'
+import React, { KeyboardEvent, RefObject, useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import styled from 'styled-components'
+import { transparentize } from 'polished'
+import { useTranslation } from 'react-i18next'
+import { FixedSizeList } from 'react-window'
+import { CloseIcon, TYPE } from '../../theme'
+import { isAddress } from '../../utils'
+import Column from '../Column'
+import { RowBetween } from '../Row'
+import { filterPairs as filterPairsBySearchQuery } from './filtering'
+import SortButton from './SortButton'
+import { usePairsComparator } from './sorting'
+import { PaddedColumn, SearchInput, Separator } from './styleds'
+import { usePairAtAddress } from '../../data/Reserves'
+import PairList from './PairList'
+import { useAllPairs } from '../../hooks/useAllPairs'
+
+interface PairSearchProps {
+  isOpen: boolean
+  onDismiss: () => void
+  selectedPair?: Pair | null
+  onPairSelect: (pair: Pair) => void
+  showCommonBases?: boolean
+  filterPairs?: (pair: Pair) => boolean
+}
+
+const Wrapper = styled.div`
+  width: 100%;
+
+  background: ${({ theme }) => transparentize(0.45, theme.bg2)};
+`
+
+export function PairSearch({ selectedPair, onPairSelect, onDismiss, isOpen, filterPairs }: PairSearchProps) {
+  const { t } = useTranslation()
+
+  const fixedList = useRef<FixedSizeList>()
+  const [searchQuery, setSearchQuery] = useState<string>('')
+  const [invertSearchOrder, setInvertSearchOrder] = useState<boolean>(false)
+  const { pairs: allPairs } = useAllPairs()
+
+  // if they input an address, use it
+  const isAddressSearch = isAddress(searchQuery)
+  const searchPair = usePairAtAddress(searchQuery)
+
+  const pairsComparator = usePairsComparator(invertSearchOrder)
+
+  const filteredPairs: Pair[] = useMemo(() => {
+    let pairs = allPairs
+    if (filterPairs) pairs = allPairs.filter(filterPairs)
+    if (isAddressSearch) return searchPair ? [searchPair] : []
+    return filterPairsBySearchQuery(pairs, searchQuery)
+  }, [allPairs, filterPairs, isAddressSearch, searchPair, searchQuery])
+
+  const filteredSortedPairs: Pair[] = useMemo(() => {
+    if (searchPair) return [searchPair]
+    const sorted = filteredPairs.sort(pairsComparator)
+    const symbolMatch = searchQuery
+      .toLowerCase()
+      .split(/\s+/)
+      .filter(s => s.length > 0)
+    if (symbolMatch.length > 1) return sorted
+
+    // sort any exact symbol matches first
+    // return sorted.filter(pair => ((pair[0].symbol || '') + (pair[1].symbol || '')).toLowerCase() === symbolMatch[0])
```

Comment:
Why is this commented out?

Reply:
I've simplified and improved the pair search experience here 1283a8fe564534b22f1b09d1a2dcaf38900bd04c